packages/nextjs/src/utils/config.ts

```diff
@@ -1,10 +1,37 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/no-var-requires */
 /* eslint-disable @typescript-eslint/no-explicit-any */
-// import { version as nextVersion } from './node_modules/next/package.json';
 import { getSentryRelease } from '@sentry/node';
 import { logger } from '@sentry/utils';
 import defaultWebpackPlugin, { SentryCliPluginOptions } from '@sentry/webpack-plugin';
 import * as SentryWebpackPlugin from '@sentry/webpack-plugin';
 import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * Starting at `startPath`, move up one directory at a time, searching for `searchFile`.
+ *
+ * @param startPath The location from which to start the search.
+ * @param searchFile The file to search for
+ * @returns The absolute path of the file, if it's found, or undefined if it's not.
+ */
+function findUp(startPath: string, searchFile: string): string | undefined {
+  if (!fs.existsSync(startPath)) {
+    throw new Error(`The given \`startPath\` value (${startPath}) does not exist.`);
+  }
+
+  let currentDir = fs.statSync(startPath).isFile() ? path.dirname(startPath) : startPath;
+  while (currentDir !== '/') {
+    const possiblePath = path.join(currentDir, searchFile);
+    if (fs.existsSync(possiblePath)) {
+      return possiblePath;
+    }
+
+    currentDir = path.join(currentDir, '..');
+  }
```

Comment:
This will break on Windows machines. Also using `path.resolve` will transform `/foo/bar/baz` into `/foo/bar`, `/foo`, `/` instead of `/foo/bar/baz/..`, `/foo/bar/baz/../..`, `/foo/bar/baz/../../..`.

```suggestion
  while (true) {
    const possiblePath = path.join(currentDir, searchFile);
    if (fs.existsSync(possiblePath)) {
      return possiblePath;
    }

    const parentDir = path.resolve(currentDir, '..');
    if (currentDir === parentDir) {
      break;
    }
    currentDir = parentDir;
  }
```

This will work on any OS and always traverse only as much as well need.

Windows:
![image](https://user-images.githubusercontent.com/1523305/115682331-a6262f00-a355-11eb-9c21-e79557b5a0b1.png)

POSIX:
![image](https://user-images.githubusercontent.com/1523305/115682423-bd651c80-a355-11eb-9ab8-f8b78f74c2ab.png)

Reply:
> Also using path.resolve will transform /foo/bar/baz into /foo/bar, /foo, / instead of /foo/bar/baz/.., /foo/bar/baz/../.., /foo/bar/baz/../../...

🤔 At least on my machine, `path.join` behaves the same way:
![image](https://user-images.githubusercontent.com/14812505/115723386-c85b8500-a334-11eb-8abf-791a1c3ce633.png)

That said, I'm realizing I do have a bug, because I'm breaking one iteration too early, and not checking the root level. And I hadn't thought about the whole windows-backslash business. Thanks for catching that.

Reply:
Oh 🤔